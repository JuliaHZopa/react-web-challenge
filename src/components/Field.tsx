import * as React from 'react';

interface IFieldProps {
  type: string;
  name: string;
  value: string | number;
  handleOnChange: (e: React.ChangeEvent) => void;
  placeholder?: string;
  label: string;
}

const Field = ({type, name, value, handleOnChange, placeholder, label}: IFieldProps) => {
  return(
    <div className='field'>
      <label>{label}</label>
      <input type={type} name={name} value={value} onChange={handleOnChange} placeholder={placeholder}/>
    </div>
  );
}

export default Field;
