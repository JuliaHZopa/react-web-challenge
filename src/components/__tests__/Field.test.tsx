import * as React from 'react';
import * as renderer from "react-test-renderer";
import Field from '../Field';

const handleOnChangeMock = jest.fn();

describe('Field', () => {
  test('it renders', () => {
    const render = renderer.create(
      <Field
        type='text'
        name='fieldName'
        value=''
        handleOnChange={handleOnChangeMock}
        label='Label'/>).toJSON();
    expect(render).toMatchSnapshot();
  });
});
