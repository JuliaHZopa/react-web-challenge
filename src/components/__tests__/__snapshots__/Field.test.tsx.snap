// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field it renders 1`] = `
<div
  className="field"
>
  <label>
    Label
  </label>
  <input
    name="fieldName"
    onChange={[MockFunction]}
    placeholder={undefined}
    type="text"
  />
</div>
`;
